pallets/runtime/tests/src/asset_test.rs

```diff
@@ -2394,6 +2394,31 @@ fn asset_type_custom_works() {
     });
 }
 
+#[test]
+fn invalid_custom_asset_type_check() {
+    ExtBuilder::default().build().execute_with(|| {
+        let owner = User::new(AccountKeyring::Dave);
+
+        // Create ticker.
+        let (ticker, _) = a_token(owner.did);
+
+        let invalid_id = CustomAssetTypeId(1_000_000);
+        assert_noop!(
+            Asset::create_asset(
+                owner.origin(),
+                ticker.as_ref().into(),
+                ticker,
+                true,
+                AssetType::Custom(invalid_id),
+                vec![],
+                None,
+                false,
+            ),
```

Comment:
Simpler if you bind `mut token`, modify `token.asset_type`, and then use `basic_asset`